### PR TITLE
Add simple adjudicator deployment

### PR DESCRIPTION
**Pull request by lalexgap**
Added a blockchain saga that will handle interacting with the blockchain. Added the ability to deploy the simple adjudicator to the blockchain as well as some crude error handling.

### CODE REVIEW

File: src/wallet/redux/actions/blockchain.ts [hunk 1]
```diff
@@ -1,22 +1,33 @@
-import { Wallet } from '../..';
+export type BlockchainDeployAdjudicatorAction = ReturnType<typeof deployAdjudicator>;
```
Comment (tomclose) [inline src/wallet/redux/actions/blockchain.ts]: Maybe call this `DeployAdjudicator` to stick with the new action pattern (you'll get the `blockchain` part from the namespace). Potentially even better would be to have something like:
```ts
type DeploymentRequest = ReturnType<typeof deploymentRequest>;
type DeploymentSuccess = ReturnType<typeof deploymentSuccess>;
type DeploymentWrongNetworkFailure = ReturnType<typeof deploymentWrongNetworkFailure>;
type DeploymentMetamaskFailure = ReturnType<typeof deploymentMetamaskFailure>;
type DeploymentFailure = DeploymentWrongNetworkFailure | DeploymentMetamaskFailure;
type DeploymentResponse = DeploymentSuccess | DeploymentFailure;
```

File: src/wallet/redux/sagas/blockchain.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// @ts-ignore
+import simpleAdjudicatorArtifact from 'fmg-simple-adjudicator/contracts/SimpleAdjudicator.sol';
+import contract from 'truffle-contract';
+import detectNetwork from 'web3-detect-network';
+// The blockchain saga will be responsible for dealing with the blockchain using truffle
+export function* blockchainSaga() {
+  const channel = yield actionChannel(blockchainActions.BLOCKCHAIN_DEPLOYADJUDICATOR);
+  while (true) {
+    const action = yield take(channel);
+    switch (action.type) {
```
Comment (tomclose) [inline src/wallet/redux/sagas/blockchain.ts]: I'm guessing the switch is here in preparation for adding more types?

File: src/wallet/redux/sagas/blockchain.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      case blockchainActions.BLOCKCHAIN_DEPLOYADJUDICATOR:
+        const simpleAdjudicatorContract = contract(simpleAdjudicatorArtifact);
+
+        const network = yield detectNetwork(web3.currentProvider);
```
Comment (tomclose) [inline src/wallet/redux/sagas/blockchain.ts]: Should this be `yield call(detectNetwork, web3.currentProvider)`?

File: src/wallet/redux/actions/blockchain.ts (hunk 1 above)
Comment (tomclose) [reply src/wallet/redux/actions/blockchain.ts]: Not entirely sure about the `Deployment*Failure` - maybe those are more general error actions. It's still useful to have that `DeploymentFailure` union type though, as you can then check you've covered all possibilities by using [the exhaustive checking pattern](https://github.com/magmo/rps-poc/blob/master/src/wallet/redux/sagas/wallet.ts#L38).

File: src/wallet/redux/sagas/blockchain.ts (hunk 3 above)
Comment (lalexgap) [reply src/wallet/redux/sagas/blockchain.ts]: Yup

File: src/wallet/redux/sagas/blockchain.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        if (!Object.keys(simpleAdjudicatorContract.networks).find(id => id === network.id)) {
```
Comment (tomclose) [inline src/wallet/redux/sagas/blockchain.ts]: This is interesting. I'm guessing this means that we're actually shipping an app running on multiple networks?

**Pull request merged**